akka-http-core/src/main/scala/akka/http/scaladsl/model/MediaType.scala

```diff
@@ -5,272 +5,253 @@
+      override def isImage = mainType == "image"
+      override def isMessage = mainType == "message"
+      override def isText = mainType == "text"
+      override def isVideo = mainType == "video"
+    }
   }
-  val `text/*` = new PredefinedMediaRange("text/*") {
-    def matches(mediaType: MediaType) = mediaType.isText
-    override def isText = true
-    def specimen = MediaTypes.`text/plain`
+
+  def customWithOpenCharset(mainType: String, subType: String, fileExtensions: List[String] = Nil,
+                            params: Map[String, String] = Map.empty,
+                            allowArbitrarySubtypes: Boolean = false): WithOpenCharset = {
+    require(mainType != "multipart", "Cannot create a MediaType.Multipart here, use `customMultipart` instead!")
+    require(allowArbitrarySubtypes || subType != "*", "Cannot create a MediaRange here, use `MediaRange.custom` instead!")
+    val _params = params
+    new NonMultipartWithOpenCharset(renderValue(mainType, subType, params), mainType, subType, fileExtensions) {
+      override def params = _params
+      override def isApplication = mainType == "application"
+      override def isAudio = mainType == "audio"
+      override def isImage = mainType == "image"
+      override def isMessage = mainType == "message"
+      override def isText = mainType == "text"
+      override def isVideo = mainType == "video"
+    }
   }
-  val `video/*` = new PredefinedMediaRange("video/*") {
-    def matches(mediaType: MediaType) = mediaType.isVideo
-    override def isVideo = true
-    def specimen = MediaTypes.`video/mp4`
+
+  def customMultipart(subType: String, params: Map[String, String]): Multipart = {
+    require(subType != "*", "Cannot create a MediaRange here, use MediaRanges.`multipart/*` instead!")
+    new Multipart(subType, params)
   }
-}
 
-sealed abstract case class MediaType private[http] (value: String)(val mainType: String,
-                                                                   val subType: String,
-                                                                   val compressible: Boolean,
-                                                                   val encoding: MediaType.Encoding,
-                                                                   val fileExtensions: immutable.Seq[String],
-                                                                   val params: Map[String, String])
-  extends jm.MediaType with LazyValueBytesRenderable with WithQValue[MediaRange] {
-  def isApplication = false
-  def isAudio = false
-  def isImage = false
-  def isMessage = false
-  def isMultipart = false
-  def isText = false
-  def isVideo = false
+  def custom(value: String, binary: Boolean, compressible: Boolean = true,
+             fileExtensions: List[String] = Nil): MediaType = {
+    val parts = value.split('/')
+    if (parts.length != 2) throw new IllegalArgumentException(value + " is not a valid media-type")
```

Comment:
Can be a bit more helpful here: 

`" is not a valid media-type. Should be 2 `/` separated parts."` (open for better wording ideas here hm)